sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataCache.java

```diff
@@ -920,7 +921,12 @@ protected Sequence<SegmentAnalysis> runSegmentMetadataQuery(
         querySegmentSpec,
         new AllColumnIncluderator(),
         false,
-        brokerInternalQueryConfig.getContext(),
+        // we are not merging segment metadata query result, why use parallel merge pool
```

Comment:
nit: "disable the parallel merge because we don't care about the merge and don't want to consume its resources"